plugins/src/lib.rs

```diff
@@ -274,7 +280,16 @@ where
             }
         }
 
-        Ok(messages::InitResponse::default())
+        match self.state.disable_reason() {
+            Some(reason) => {
+                let init_resp = messages::InitResponse{
+                    disable: Some(reason),
+                };
+                Ok(init_resp)
+            },
+            None => Ok(messages::InitResponse::default()),
+        }
+
```

Comment:
auto review, this simplification should work

```suggestion
    let init_resp = messages::InitResponse{
             disable: self.state.disable_reason(),
           };
    Ok(init_resp)

```